add estimation period for quantity

leerogers-bjss opened the pull request:
Rename estimation period on order item price to billing period.

update order form quantity pages to display different content depending on the estimation period.

Add RangeDefinition logic to flat price tests.

Add SetOrderItemEstimation Tests

leerogers-bjss committed:
Add in logic to set esitmaiton period on order item.

Rename estimation period on order item price to billing period.

update order form quantity pages to display different content depending on the estimation period.

leerogers-bjss committed:
Post merge fix issues.

Fix references to estimationPeriod back to BIlling Period

Remove SelectedCalculationType Being Hidden

Add RangeDefinition logic to flat price tests.

Add SetOrderItemEstimation Tests

bjss-colin commented on src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/ListPrices/AddEditFlatListPrice.cshtml:
```diff
@@ -121,6 +121,11 @@
                 character-count="false" />
 
+            <nhs-input
+                label-text="Range definition"
```
Range definition isn't the correct term here for the label text as a flat price has no ranges. Might be worth running this by Luke to coin up a better term.

magnusjohn-bjss replied:
Not sure I agree with this, range definition is the term we use for the plural of the unit description (i.e. "beds" when the unit is "per bed"), it has descriptive lede text to make that clear and should be consistent between flat & tiered prices.

bjss-colin reviewed (approved):
Approved for now but it's worth re-reviewing the `Range definition` term. I still don't think it's the right term here.

leerogers-bjss committed:
Content Update from luke

Changed flat price range definition to units with better hint-text.

rerodered admin list prices pages to but optional field of unit definiton below the required fields.

leerogers-bjss committed:
Change test order seed data to insert into BillingPeriodId not EstimationPeriodId on OrderItemPrices

leerogers-bjss committed:
Updated OrderItemPrice Billing Period Field to not point at EstimationPeriodId column but BillingPeriodId Column

leerogers-bjss committed:
Remove Units from being part of Duplication check

Change Units Required Error message to not mention range definition

Update tests to reflect changes

leerogers-bjss committed:
Fix bad reference on AddEditFlatListPriceModelValidatorTests

magnusjohn-bjss committed:
Test fix
